default.nix

```diff
@@ -0,0 +1,122 @@
+  };
+
+  spago-overlay = final: prev: {
+
+    myhaskell = prev.haskell // {
+      packages = prev.haskell.packages // {
+        native-bignum = prev.haskell.packages.native-bignum // {
+          ghc902 = prev.haskell.packages.native-bignum.ghc902.override (oldAttrs: {
+            overrides =
+              final.lib.composeExtensions
+                (oldAttrs.overrides or (_: _: {}))
+                (hfinal: hprev: {
+
+                  aeson = hfinal.aeson_1_5_6_0;
+
+                  # bower-json needs aeson_1_5_6_0 and is marked broken without it.
+                  bower-json = final.haskell.lib.doDistribute (final.haskell.lib.markUnbroken hprev.bower-json);
+
+                  spago =
+                    let
+                      filesToIgnore = [
+                        "default.nix"
+                        ".git"
+                        ".github"
+                        "Makefile"
+                        "result"
+                        ".stack-work"
+                        "stack.yaml"
+                      ];
+
+                      src = builtins.path {
+                        name = "spago-src";
+                        path = ./.;
+                        filter = path: type:
+                          with final.lib;
+                          ! elem (baseNameOf path) filesToIgnore &&
+                          ! any (flip hasPrefix (baseNameOf path)) [ "dist" ".ghc" ];
+                      };
+
+
+                      docsSearchApp_0_0_10 = final.fetchurl {
+                        url = "https://github.com/purescript/purescript-docs-search/releases/download/v0.0.10/docs-search-app.js";
+                        sha256 = "0m5ah29x290r0zk19hx2wix2djy7bs4plh9kvjz6bs9r45x25pa5";
+                      };
+
+                      docsSearchApp_0_0_11 = final.fetchurl {
+                        url = "https://github.com/purescript/purescript-docs-search/releases/download/v0.0.11/docs-search-app.js";
+                        sha256 = "17qngsdxfg96cka1cgrl3zdrpal8ll6vyhhnazqm4hwj16ywjm02";
+                      };
+
+                      purescriptDocsSearch_0_0_10 = final.fetchurl {
+                        url = "https://github.com/purescript/purescript-docs-search/releases/download/v0.0.10/purescript-docs-search";
+                        sha256 = "0wc1zyhli4m2yykc6i0crm048gyizxh7b81n8xc4yb7ibjqwhyj3";
+                      };
+
+                      purescriptDocsSearch_0_0_11 = final.fetchurl {
+                        url = "https://github.com/purescript/purescript-docs-search/releases/download/v0.0.11/purescript-docs-search";
+                        sha256 = "1hjdprm990vyxz86fgq14ajn0lkams7i00h8k2i2g1a0hjdwppq6";
+                      };
+
+                      spagoDrvFromCabal2nix =
+                        final.haskell.lib.overrideCabal
+                          (hfinal.callCabal2nix "spago" src {})
+                          (drv: {
+                            doCheck = false;
```

Comment:
Since `spago` uses the network during tests, I've had to disable the tests here.

We end up not really testing this Nix-built `spago` binary, which is unfortunate.  We could potentially add some simple tests to the `build.yaml` CI file, like running `spago version`, `spago init`, `spago build`, etc.